aggregator/src/aggregator.rs

```diff
@@ -465,6 +465,41 @@ impl<C: Clock> Aggregator<C> {
             .await
     }
 
+    async fn handle_aggregate_delete(
+        &self,
+        task_id: &TaskId,
+        aggregation_job_id: &AggregationJobId,
+        auth_token: Option<AuthenticationToken>,
+        taskprov_task_config: Option<&TaskConfig>,
+    ) -> Result<(), Error> {
+        let task_aggregator = self
+            .task_aggregator_for(task_id)
+            .await?
+            .ok_or(Error::UnrecognizedTask(*task_id))?;
+        if task_aggregator.task.role() != &Role::Helper {
+            return Err(Error::UnrecognizedTask(*task_id));
+        }
```

Comment:
Like the other aggregation job request paths, DELETE is only supported on the helper. We could also allow it on the leader, since it also has an `aggregation_jobs` table with rows that could be marked deleted. That could be useful in the context of job queue management which we have discussed in broad terms recently. However I'd prefer not to support DELETE on the leader until we have a more coherent plan for what to use that for.